tracer/src/Datadog.Trace/DuckTyping/DuckTypeExceptions.cs

```diff
@@ -177,16 +177,16 @@ internal static void Throw(FieldInfo field)
     /// </summary>
     internal class DuckTypePropertyOrFieldNotFoundException : DuckTypeException
     {
-        private DuckTypePropertyOrFieldNotFoundException(string name, string duckAttributeName)
-            : base($"The property or field '{duckAttributeName}' for the proxy property '{name}' was not found in the instance.")
+        private DuckTypePropertyOrFieldNotFoundException(string name, string duckAttributeName, string type)
+            : base($"The property or field '{duckAttributeName}' for the proxy property '{name}' was not found in the instance of type '{type}'.")
```

Comment:
Nice addition 👍 You might want to pull these unrelated tweaks out into a separate PR for simplicity, up to you 🙂